### ISSUE TITLE
Rails::Generators::Actions#gem generates a line against POSIX standard definition of a line

### ISSUE THREAD

**Issue by masakazutakewaka** [rails/rails#35284]
### Steps to reproduce
I came across this problem when I used https://github.com/rmosolgo/graphql-ruby so let me showcase the steps by following exactly what I did with this gem.

```
$ rails new sample
$ cd sample
# add graphql-ruby to your Gemfile
$ bundle
$ bundle exec rails g graphql:install
```

One of the many things that `bundle exec rails g graphql:install` does is to add `gem 'graphiql-rails'` to the end of your Gemfile.

### Expected behavior
The last line we added should have `\n` at the end.

```
# in your rails project
$ cat Gemfile
 .
 .
 .

# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
gem 'graphiql-rails', group: :development
```

### Actual behavior
The last line doesn't have `\n` at the end which is against the way the POSIX defines a line.

Gemfile
```
# in your rails project
$ cat Gemfile
.
.
.

# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
gem 'graphiql-rails', group: :development%
```

### System configuration
**Rails version**:
5.2.2
**Ruby version**:
2.6.1

**Comment by masakazutakewaka** [comment]
So I looked into it and found out that the problem is in `Rails::Generators::Actions#gem`

https://github.com/rails/rails/blob/b414ca3f17c76fb1cc43d79fbffd2f3ed95acdde/railties/lib/rails/generators/actions.rb#L40-L45

```
        in_root do
          str = "gem #{parts.join(", ")}"
          str = indentation + str
          str = "\n" + str
          append_file "Gemfile", str, verbose: false
        end
```

The reason why it doesn't have a newline at the end is that you put a newline on top of the variable `str`.

**Comment by masakazutakewaka** [comment]
This is my small patch that adds a newline at the end of a line. https://github.com/rails/rails/pull/35285

Obviously this change does not quite solve the problem especially in a case where you use `Rails::Generators::Actions#gem` multiple times on your Gemfile.

So what should we do about this? Any feedbacks or takes on this problem is much appreciated.

At the same time I do recognize leaving this alone is also an option as it doesn't break things so easily.

**Comment by y-yagi** [comment]
Fixed by #35285.
